app/queries/servers/user.ts

```diff
@@ -110,9 +112,17 @@ export const observeUserIsTeamAdmin = (database: Database, userId: string, teamI
 
 export const observeUserIsChannelAdmin = (database: Database, userId: string, channelId: string) => {
     const id = `${channelId}-${userId}`;
-    return database.get<ChannelMembershipModel>(CHANNEL_MEMBERSHIP).query(
+    const myChannelRoles = observeMyChannel(database, channelId).pipe(
+        switchMap((mc) => of$(mc?.roles || '')),
+        distinctUntilChanged(),
+    );
+    const channelSchemeAdmin = database.get<ChannelMembershipModel>(CHANNEL_MEMBERSHIP).query(
```

Comment:
I doubt this is really needed as the current user will also have a record in the `ChannelMembership` table, thus the `schemeAdmin` should be set as true there as well.

Can you let me know what did you see that made you go this route?

Reply:
@enahum Sure: 
1. have a regular user create a public channel
2. in that channel, user should be able to disable calls. (only channel admins or system admins can do so)

This works the first time they create the channel. But as soon as the websocket loses connection and reconnects, or you log out and log back in, 2 doesn't work.

Reply:
@cpoile but the thing is that myChannel is only applicable to the current user, so that is what I'm not understanding

Reply:
@enahum The current user is the one that we're checking whether or not they can disable calls -- or do you mean something else?

Reply:
> @enahum The current user is the one that we're checking whether or not they can disable calls -- or do you mean something else?

so checking the schemeAdmin in the ChannelMembership table should be enough, if that is not the case, then probably we have an issue elsewhere that is not setting the correct value.

We have two alternatives, we leave this as is so that it falls back to use the roles for the current user, or we try and figure out why the `schemeAdmin` value is not set correctly when the regular user creates a public channel.

your call.

Reply:
As we discussed, filed a bug: https://mattermost.atlassian.net/browse/MM-46418
Will keep this patch until that fix removes it.